drivers/scheduler/k8s/k8s.go

```diff
@@ -7517,3 +7528,29 @@ func ClusterVersion() (string, error) {
 	}
 	return strings.TrimLeft(ver.String(), "v"), nil
 }
+
+func createClonedStorageClassIfRequired(originalStorageClass *storageapi.StorageClass) (string, error) {
+	cloneStorageClass := originalStorageClass
+	clonedSCName := cloneStorageClass.Name + "-clone"
+	clonedSCobj, err := k8sStorage.GetStorageClass(clonedSCName)
+	if err == nil {
+		return clonedSCobj.Name, nil
+	} else if strings.Contains(err.Error(), fmt.Sprintf("storageclasses.storage.k8s.io \"%v\" not found", clonedSCName)) {
+		log.Infof("Cloned SC with the name %v does not exist and will be created", clonedSCName)
+
+		cloneStorageClass.Name = clonedSCName
```

Comment:
Has this been tested? 
If err isn't nil, the object `clonedStorageClass` is likely nil, and we will end up with a kernel panic.

Reply:
Yes, I tested it. I have attached a jenkins job link in the description above.

Reply:
Removing reference to `clonedSCobj` when `err != nil` frees us from the risk of a kernel panic, however shall we still include the storage class name in this log?  Can simply use `clonedSCName`

Reply:
done